### PR TITLE
EditOnline Button in Preview

### PR DESCRIPTION
**Pull request by fsabbek**
Check EditOnline Button in Preview

### CODE REVIEW

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java [hunk 1]
```diff
@@ -30,13 +30,12 @@ public OnlyOfficeEditingPage(TestBase testBase) {
...
      */
-    public void checkOpeningDocumentWithEditOnline(String documentName, String userName) {
-        $(byText(documentName + ".docx")).waitUntil(visible, openBrowserTimeoutMs);
+    public void checkOpeningDocumentWithEditOnline(String documentName,String extension) {
+        $(byText(documentName + extension)).waitUntil(visible, openBrowserTimeoutMs);
         $(".editor").waitUntil(visible, 60000).exists();
         switchTo().frame("frameEditor");
-        $("#title-user-name").waitUntil(exactText(userName), 120000);
-        $("#id_panel_left").waitUntil(visible, 120000);
+        $("#view-left-menu").parent().find(".tool-menu.left").waitUntil(visible, 120000);
```
Comment (thomasdelhomenie) [inline onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: Why did you remove the check on title-user-name element ? It seems to be relevant to check that the user name is correctly displayed.

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java [hunk 2]
```diff
@@ -30,13 +30,12 @@ public OnlyOfficeEditingPage(TestBase testBase) {
     /**
      * Check that document is opened on OnlyOffice Editor
      * @param documentName
-     * @param userName
+     * @param extension
      */
-    public void checkOpeningDocumentWithEditOnline(String documentName, String userName) {
-        $(byText(documentName + ".docx")).waitUntil(visible, openBrowserTimeoutMs);
+    public void checkOpeningDocumentWithEditOnline(String documentName,String extension) {
```
Comment (thomasdelhomenie) [inline onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: Why did you split the doc name ? It seems the extension is never use alone.

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java [hunk 3]
```diff
@@ -36,7 +43,21 @@ public void editOnlineFromAS() {
...
+     * @param element1
+     * @param element2
+     * @param element3
+     */
+    public void checkAlignmentElement(SelenideElement element1, SelenideElement element2, SelenideElement element3) {
+        int locElement1Button = element1.getLocation().y;
+        int locElement2Button = element2.getLocation().y;
+        int locElement3Button = element3.getLocation().y;
+        assertEquals(locElement1Button,locElement2Button,locElement3Button);
+    }
```
Comment (thomasdelhomenie) [inline onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java]: I don't think it is the role of the functional test to check the pixel position of elements (this must be done by BackstopJS tests). We should only check that the buttons are displayed IMO.

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
 import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
 import static com.codeborne.selenide.Selenide.$;
 
 public class OnlyOfficeLocator {
 
     public static final SelenideElement ELEMENT_EDIT_ONlINE_BUTTON = $(".uiIconEcmsOnlyofficeOpen.uiIconEcmsLightGray.uiIconEdit");
     public static final SelenideElement ELEMENT_EDIT_ONLINE_BUTTON_FROM_PREVIEW = $(".onlyOfficeEditBtn");
+    public static final SelenideElement ELEMENT_DOWNLOAD_BUTTON_FROM_AS_TEXT = $(By.xpath("((//*[@class='pull-left statusAction'])[1]//a)[1]"));
+    public static final SelenideElement ELEMENT_OPEN_BUTTON_FROM_AS_TEXT = $(By.xpath("((//*[@class='pull-left statusAction'])[1]//a)[2]"));
```
Comment (thomasdelhomenie) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: No need to check the buttons Download and Open IMO, there are not part of the feature. We should only check if the Edit Online button is displayed.

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
+    public static final SelenideElement ELEMENT_EDIT_ONLINE_BUTTON_FROM_AS_TEXT = $(By.xpath("((//*[@class='pull-left statusAction'])[1]//a)[3]"));
```
Comment (thomasdelhomenie) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: Instead of using xpath we should add a class on the button so you can select the element in a more robust way.

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java (hunk 1 above)
Comment (fsabbek) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: I'm checking the opening, and the element I used before does not work with l pptx so I had to change a deeper element for the 3 file types, more than the user name

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java (hunk 2 above)
Comment (fsabbek) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: for "checkButtonDocument(document)" I use only the name of document for US1 and US2, and I think it's more clear for all types

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java (hunk 3 above)
Comment (fsabbek) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java]: but I used the pixel to check just horizontal position (alignment),  also when I did the research I found that it is often used

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java (hunk 4 above)
Comment (fsabbek) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: In US3 I should check the alignment of Edit Online button with other buttons!! so I should compare edit online to the other buttons

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java (hunk 5 above)
Comment (fsabbek) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: I waiting for this comment :), just for this case I Used Xpath because I just need the alignment of elements of "pull left statusAction" and I compared with the next elements, also for download button I cant use ccs selector easily.

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java (hunk 1 above)
Comment (thomasdelhomenie) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: ok but can we keep the check on the username as well ?

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java (hunk 3 above)
Comment (thomasdelhomenie) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java]: you mean it is often used in our selenium tests ?
That's not the job of these tests IMO.

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java (hunk 4 above)
Comment (thomasdelhomenie) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: You do not have to check the alignment, only the existence (and maybe the order).

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java (hunk 5 above)
Comment (thomasdelhomenie) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: > I waiting for this comment :)

:smile: 

> just for this case I Used Xpath because I just need the alignment of elements of "pull left statusAction" and I compared with the next elements, also for download button I cant use ccs selector easily.

The XPath is really fragile. If we add a button at the beginning for example, the test fails, even if OnlyOffice still work correctly. That's why we should only check Edit Online existence and add a class name on the element in the OnlyOffice code to select it easily.

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java (hunk 1 above)
Comment (fsabbek) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: I will check if it the same element of all types

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java (hunk 4 above)
Comment (fsabbek) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: US3 :
The "edit online" button should be aligned with other buttons in different locations:
-AS
-documents app
-file's preview

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java (hunk 5 above)
Comment (fsabbek) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/onlyOffice/OnlyOfficeLocator.java]: But the existing of Edit Online is checked in US1 and US2, This US is about alignment and I checked also with Sameh. it's about alignment.

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java (hunk 1 above)
Comment (thomasdelhomenie) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeEditingPage.java]: :+1:

File: onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java (hunk 3 above)
Comment (fsabbek) [reply onlyOffice/src/main/java/org/exoplatform/platform/qa/ui/onlyOffice/pageobject/OnlyOfficeActivityStream.java]: no no, it never used in our selenium
